applications/ParticleMechanicsApplication/tests/cpp_tests/test_search_element_utility.cpp

```diff
@@ -470,80 +470,82 @@ namespace Testing
-        PrepareModelPart(r_mpm_model_part, r_background_model_part, mp_coordinate, int_weight);
-
-        r_background_model_part.GetProcessInfo().SetValue(IS_PQMPM, true);
-        r_background_model_part.GetProcessInfo().SetValue(IS_PQMPM_FALLBACK_TO_MPM, false);
-
-        r_mpm_model_part.GetElement(2).SetValuesOnIntegrationPoints(
-            MP_COORD, { mp_coordinate }, r_mpm_model_part.GetProcessInfo());
-        r_mpm_model_part.GetElement(2).SetValuesOnIntegrationPoints(
-            MP_VOLUME, mp_volume, r_mpm_model_part.GetProcessInfo());
-        #pragma omp single
-        KRATOS_CHECK_EXCEPTION_IS_THROWN(MPMSearchElementUtility::SearchElement<2>(
-            r_background_model_part, r_mpm_model_part, 1000, 1e-6) ,
-            "Error: ERROR")
-    }
-    /*
-    /// PQMPM test 7 - Check pqmpm fails if point tries to split outside mesh
-    KRATOS_TEST_CASE_IN_SUITE(MPMSearchElementPQMPM2DOutsideSplit, KratosParticleMechanicsFastSuite)
-    {
-        const double int_weight_test_tolerance = 1e-5;
+    // TEST DISABLED - KRATOS_CHECK_EXCEPTION_IS_THROWN doesn't work in parallel!
+    ///// PQMPM test 6 - Check pqmpm fails with unstructured 3D mesh
+    //KRATOS_TEST_CASE_IN_SUITE(MPMSearchElementPQMPM3DHexError, KratosParticleMechanicsFastSuite)
+    //{
+    //    const double int_weight_test_tolerance = 1e-5;
+    //
+    //    // First Coordinates of Material Point
+    //    array_1d<double, 3> mp_coordinate;
+    //    mp_coordinate[0] = 0.8;
+    //    mp_coordinate[1] = 0.8;
+    //    mp_coordinate[2] = 0.5;
+    //    const double int_weight = 1.0;
+    //    std::vector<double> mp_volume(1);
+    //    mp_volume[0] = 0.8;
+    //
+    //    // Case of background grid( 0=quad, 10=tri, 20=hex. +1 = skew)
+    //    const IndexType grid_case = 21;
+    //
+    //    Model current_model;
+    //    ModelPart& r_mpm_model_part = current_model.CreateModelPart("MPMModelPart");
+    //
+    //    ModelPart& r_background_model_part = current_model.CreateModelPart("MPMBackgroundModelPart");
+    //    PrepareGeneralBackgroundModelPart(r_background_model_part, grid_case);
+    //    PrepareModelPart(r_mpm_model_part, r_background_model_part, mp_coordinate, int_weight);
+    //
+    //    r_background_model_part.GetProcessInfo().SetValue(IS_PQMPM, true);
+    //    r_background_model_part.GetProcessInfo().SetValue(IS_PQMPM_FALLBACK_TO_MPM, false);
+    //
+    //    r_mpm_model_part.GetElement(2).SetValuesOnIntegrationPoints(
+    //        MP_COORD, { mp_coordinate }, r_mpm_model_part.GetProcessInfo());
+    //    r_mpm_model_part.GetElement(2).SetValuesOnIntegrationPoints(
+    //        MP_VOLUME, mp_volume, r_mpm_model_part.GetProcessInfo());
+    //    #pragma omp single
+    //    KRATOS_CHECK_EXCEPTION_IS_THROWN(MPMSearchElementUtility::SearchElement<2>(
+    //        r_background_model_part, r_mpm_model_part, 1000, 1e-6) ,
+    //        "Error: ERROR")

```

Comment:
Hi @philbucher , is there a way to check exceptions that are thrown within a parallel loop? Or to make a loop serial during cpp tests only? A similar problem was left unsolved in structural mechanics app [here](https://github.com/KratosMultiphysics/Kratos/blob/c3e91a1964ab2c4eee1f822986a67e38f452a4f8/applications/StructuralMechanicsApplication/tests/cpp_tests/test_replace_multiple_elements_and_conditions_process.cpp#L96)

Reply:
long story short: with OpenMP: No, with C++11 thread-based shared-mem parallelism: Yes
In the long run we will probably move away from OpenMP (see #6767)

I can explain more in detail when we speak next time